examples/src/test/kotlin/DistinctTest.kt

```diff
@@ -0,0 +1,82 @@
+import com.mongodb.client.model.Filters
+import com.mongodb.kotlin.client.coroutine.MongoClient
+import io.github.cdimascio.dotenv.dotenv
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import kotlin.test.assertEquals
+
+
+class DistinctTest {
+
+    // :snippet-start: example-data-class
+    data class Movie(
+        val type: String,
+        val languages: List<String>,
+        val countries: List<String>,
+        val awards: Awards){
+            data class Awards(val wins: Int)
+        }
+    // :snippet-end:
+
+    companion object {
+        val dotenv = dotenv()
+        val client = MongoClient.create(dotenv["MONGODB_CONNECTION_URI"])
+        val database = client.getDatabase("sample_mflix")
+        val collection = database.getCollection<Movie>("movies")
+
+        @BeforeAll
+        @JvmStatic
+        fun beforeAll() {
+            runBlocking {
+                val movies = listOf(
+                    Movie("movie", listOf("English", "French"), listOf("USA", "France"), Movie.Awards(1)),
+                    Movie("movie", listOf("English", "German"), listOf("USA", "Germany"), Movie.Awards(2)),
+                    Movie("movie", listOf("English"), listOf("USA", "Australia"), Movie.Awards(3)),
+                )
+                collection.insertMany(movies)
+            }
+        }
+
+        @AfterAll
+        @JvmStatic
+        fun afterAll() {
+            runBlocking {
+                database.drop()
+                client.close()
+            }
+        }
+    }
+
+    @Test
```

Comment:
nit: for these tests, rather than the names being about the subject matter, could you make them about what the _behavior_ is that's being tested? also update bluehawk snippet as well pls

so for example, for this test, rather than being about countries, you could call it "simple distinct". for ex: 

```kt
    @Test
    fun simpleDistinctTest() = runBlocking {
        val distinct =
            // :snippet-start: simple-distinct
            collection.distinct<String>(Movie::countries.name)
            // :snippet-end:
                .toList()
        assertEquals(4, distinct.size)
    }
```

the advantages of doing this are:

1. dont need to rename test/snippet if the example subject changes
2. makes it clearer what conceptually is being tested here

Reply:
Tests renamed